deep-learning/src/main/scala/com/microsoft/ml/spark/onnx/ONNXModel.scala

```diff
@@ -262,16 +264,32 @@ object ONNXModel extends ComplexParamsReadable[ONNXModel] with Logging {
     }
   }
 
-  private def flattenNested[T: ClassTag](nestedSeq: Seq[_]): Seq[T] = {
-    nestedSeq.flatMap {
-      case x: T => Array(x)
+  private def writeNestedSeqToBuffer[T: ClassTag](nestedSeq: Seq[_], bufferWrite: T => Unit): Unit = {
+    nestedSeq.foreach {
+      case x: T => bufferWrite(x)
       case s: Seq[_] =>
-        flattenNested(s)
-      case a: Array[_] =>
-        flattenNested(a)
+        writeNestedSeqToBuffer(s, bufferWrite)
     }
   }
 
+  private def writeNestedSeqToStringBuffer(nestedSeq: Seq[_], size: Int): ArrayBuffer[String] = {
```

Comment:
Could you share a little details on why this stuff is necessary?

Reply:
This method `writeNestedSeqToStringBuffer`, together with `writeNestedSeqToBuffer`, are used to replace the `flattenNested` method.

`flattenNested` flattens a nested Seq to a 1-d array, but is super slow due to repeated memory allocation.
`writeNestedSeqToStringBuffer` and `writeNestedSeqToBuffer` both allocates the memory buffer one time, and write the nested `Seq` to the buffer. I did some local testing, and they appear to be 5-10 times faster than `flattenNested`.

The reason I need two methods is because there is not a nio.Buffer class for string type, so I had to replace it with a ArrayBuffer[String].